### PR TITLE
Remove fetching toasts

### PR DESCRIPTION
**Pull request by atappy**
in this PR I started by just removing the 'fetching' toasts roaming here and there.

Then I implemented a way to share a QR code as an image using other apps. I did it using the FileProvider class (doc available here : https://developer.android.com/reference/android/support/v4/content/FileProvider) and adapted it to our needs, centralizing it in an object named QRCode, which you can simply feed with a qrcode bitmap and the parent activity. As scanning is hard to test, I didn't manage to do automated tests but simply did it manually.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -77,6 +78,17 @@
...
+
+        <provider
+            android:name="androidx.core.content.FileProvider"
+            android:authorities="com.palfinder.fileprovider"
+            android:exported="false"
+            android:grantUriPermissions="true">
+            <meta-data
+                android:name="android.support.FILE_PROVIDER_PATHS"
+                android:resource="@xml/file_paths" />
+        </provider>
```
Comment (TruddyTheDuddi) [inline app/src/main/AndroidManifest.xml:90]: What does this do exactly, could you explain me?

**Review by TruddyTheDuddi — APPROVED** [review]
Clean and well commented code, especially the QRCode object, nice job!

File: app/src/main/java/com/github/palFinderTeam/palfinder/utils/image/QRCode.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import java.io.IOException
+
+
+object QRCode {
+    /**
+     * Saves the image as PNG to the app's private external storage folder.
+     * @param image Bitmap to save.
+     * @return Uri of the saved file or null
+     */
+    private fun saveImageExternal(image: Bitmap, parent: Activity): Uri? {
```
Comment (TheblueMan003) [inline app/src/main/java/com/github/palFinderTeam/palfinder/utils/image/QRCode.kt:24]: Some test could be added for this class

**Review by TheblueMan003 — APPROVED** [review]
Otherwise looks good

File: app/src/main/java/com/github/palFinderTeam/palfinder/utils/image/QRCode.kt (hunk 2 above)
Comment (atappy) [reply app/src/main/java/com/github/palFinderTeam/palfinder/utils/image/QRCode.kt:24]: yes I will try to add it in my next PR !

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (atappy) [reply app/src/main/AndroidManifest.xml:90]: It is the new way to access the local files on an android phone, as explained here : https://developer.android.com/reference/android/support/v4/content/FileProvider

**Pull request merged**